blots/scroll.ts

```diff
@@ -164,10 +164,10 @@ class Scroll extends ScrollBlot {
     const last = lines.pop();
     let after;
     if (last != null) {
-      after = child.split(offset + first.delta.length());
+      after = child.split(offset + first.delta.length(), true);
       Object.keys(first.attributes).forEach(name => {
         // @ts-ignore
-        child.format(name, first.attributes[name]);
+        after.prev.format(name, first.attributes[name]);
```

Comment:
`child` may have the same reference as `after` when `offset + first.delta.length()` is 0. We use `after.prev` to make sure we always refer to the right node.